### PR TITLE
Integrating environment creation with POST endpoint

### PR DESCRIPTION
**Pull request by aaronabraham311**
- Added friendly name
- Adding environment now posts to backend and updates state

Kapture 2021-04-25 at 09 59 51

### CODE REVIEW

File: src/lib/dataAccess.js
Comment (aaronabraham311) [inline src/lib/dataAccess.js:24]: Interesting issue that I found. Axios seems to set a default header for `content-type`, which the backend does not allow (only allows `x-requested-with` from what I understand). I had to manually set empty headers but perhaps this may require some backend/frontend changes. Thoughts @jaydhulia @ahmedhamodi

Comment (ahmedhamodi) [reply src/lib/dataAccess.js:24]: I'm okay to set empty headers, but this is def more of a Jay question so will defer to him.

File: src/lib/endpoints.js
Comment (ahmedhamodi) [inline src/lib/endpoints.js:9]: I think we should move `/api/admin/v1/` to a env variable so that if we change to v2 in the future, it's an easy change.

Can you also retroactively make this change to the `getAllScenarios`?

**Review by ahmedhamodi — COMMENTED** [review]
Will approve once we resolve the empty headers comment. Otherwise, small comments from me.

File: src/lib/dataAccess.js
Comment (jaydhulia) [reply src/lib/dataAccess.js:24]: We can update the backend’s allowed headers to include this header (content-type), it’s a pretty standard header. We will also need to update this PR (and the backend allowed headers) to allow CSRF token once that’s merged in

Comment (aaronabraham311) [reply src/lib/dataAccess.js]: Addressed content-type with a proposed change here: https://github.com/uwblueprint/cdc-backend/pull/84

@ahmedhamodi should I address CSRF now or only when Jay merges?

File: src/Components/Admin/admin.js
Comment (jaydhulia) [inline src/Components/Admin/admin.js:147]: Is error handling and character restrictions (on friendly name for example) going to be a future PR?

Comment (aaronabraham311) [reply src/Components/Admin/admin.js:147]: Yes, not a priority right now.

File: src/lib/dataAccess.js
Comment (ahmedhamodi) [reply src/lib/dataAccess.js]: I'm actually not sure what work is needed to address CSRF - if it's minimal I say go for it now @aaronabraham311

Comment (aaronabraham311) [reply src/lib/dataAccess.js]: Let's do it once Jay merges his PR because I'm not quite sure what I would need to do as of now. Would like this merged as an example since @andstun will be working on this soon

**Review by ahmedhamodi — APPROVED** [review]
Changes lgtm!

**Pull request merged**
